contracts/base/LiquidityManagement.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {ILockCallback} from "@uniswap/v4-core/contracts/interfaces/callback/ILockCallback.sol";
+import {TickMath} from "@uniswap/v4-core/contracts/libraries/TickMath.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {IPoolManager} from "@uniswap/v4-core/contracts/interfaces/IPoolManager.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/contracts/types/PoolId.sol";
+
+import {LiquidityAmounts} from "../libraries/LiquidityAmounts.sol";
+import {PeripheryImmutableState} from "./PeripheryImmutableState.sol";
+import {PeripheryPayments} from "./PeripheryPayments.sol";
+
+/// @title Liquidity management functions
+/// @notice Internal functions for safely managing liquidity in Uniswap V4
+abstract contract LiquidityManagement is ILockCallback, PeripheryImmutableState, PeripheryPayments {
+    using CurrencyLibrary for Currency;
+    using PoolIdLibrary for PoolKey;
+
+    error PriceSlippage();
+
+    enum CallbackType {AddLiquidity}
+
+    struct CallbackData {
+        CallbackType callbackType;
+        address sender;
+        bytes params;
+    }
+
+    struct AddLiquidityParams {
+        PoolKey poolKey;
+        int24 tickLower;
+        int24 tickUpper;
+        uint256 amount0Desired;
+        uint256 amount1Desired;
+        uint256 amount0Min;
+        uint256 amount1Min;
+        bytes hookData;
+    }
+
+    /// @notice Add liquidity to an initialized pool
+    function addLiquidity(AddLiquidityParams memory params)
+        internal
+        returns (uint128 liquidity, uint256 amount0, uint256 amount1)
+    {
+        (liquidity, amount0, amount1) = abi.decode(
+            poolManager.lock(abi.encode(CallbackData(CallbackType.AddLiquidity, msg.sender, abi.encode(params)))),
+            (uint128, uint256, uint256)
+        );
+    }
+
+    function addLiquidityCallback(AddLiquidityParams memory params)
```

Comment:
addLiquidityCallback is a strange name for an internal handler function because callbacks imply an external caller. Also for internal functions we can use the _

maybe `_addLiquidity`?